cpp/include/cudf/dictionary/dictionary_column_view.hpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column_view.hpp>
+#include <cudf/column/column.hpp>
+
+namespace cudf
+{
+
+/**
+ * @brief A wrapper class for operations on a dictionary column.
+ *
+ * A dictionary column contains a set of keys and a column of indices.
+ * The keys are a sorted set of unique values for the column.
+ * The indices values are position indices into the keys.
+ */
+class dictionary_column_view : private column_view
+{
+public:
+    dictionary_column_view( column_view const& dictionary_column );
+    dictionary_column_view( dictionary_column_view&& dictionary_view ) = default;
+    dictionary_column_view( const dictionary_column_view& dictionary_view ) = default;
+    ~dictionary_column_view() = default;
+    dictionary_column_view& operator=(dictionary_column_view const&) = default;
+    dictionary_column_view& operator=(dictionary_column_view&&) = default;
+
+    using column_view::size;
+    using column_view::null_mask;
+    using column_view::null_count;
+    using column_view::has_nulls;
+    using column_view::offset;
+
+    /**
+     * @brief Returns the parent column.
```

Comment:
What is a parent column?

Reply:
There are three columns involved here. A dictionary column (parent) and its two children: keys and indices. The parent column itself holds the nulls, offset, size, etc and so needs to be accessible when creating a `column_device_view` for example. The wrapper class here provides convenient accessors to the components of a generic `column` of type dictionary.

Reply:
So the parent is the column viewed by this view? Is parent a good name for this? It implies that a view is a child of a column, which it is not...